### PR TITLE
Refactor Window storage to use sfl::static_vector

### PR DESCRIPTION
**Pull request by ZehMatt**
Puts the Windows in their own storage rather than relying on the external memory. I've kept the changes relatively simple for now to just have it in a working condition before doing further refactors.

### CODE REVIEW

File: src/OpenLoco/src/Ui/WindowManager.cpp [hunk 1]
```diff
@@ -488,22 +490,12 @@ namespace OpenLoco::Ui::WindowManager
...
-        for (Ui::Window* w = &_windows[0]; w != _windowsEnd; w++)
-        {
-            if (w == &pWindow)
-                return i;
-
-            i++;
-        }
-
-        return 0;
+        return &pWindow - _windows.data();
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Ui/WindowManager.cpp:493]: Just for my understanding, would this not need to be divided by `sizeof(Window)`?

**Review by AaronVanGeffen — APPROVED** [review]
Nice work! Appears to be working well, and handles the current window limit fine.

With this in, I'd like to increase the window limit from 12 to -say- 50. That can be done in a separate commit in this PR or in a follow-up PR. Up to you. Doing so would finally close issue #2 (yes, two!).

File: src/OpenLoco/src/Ui/WindowManager.cpp (hunk 1 above)
Comment (ZehMatt) [reply src/OpenLoco/src/Ui/WindowManager.cpp:493]: No, it gives the distance by pointer type and not by bytes.

Comment (AaronVanGeffen) [reply src/OpenLoco/src/Ui/WindowManager.cpp:493]: Alright, nice.

**Pull request merged**